From 1d7031dde8da39cd2151f87d68dddea91f304603 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Sat, 19 Jun 2021 11:31:35 +0200
Subject: [PATCH] add --disable-example

Add --disable-example to be able to build daq on toolchains without
threads support

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Upstream status: https://github.com/snort3/libdaq/pull/13]
---
 Makefile.am  | 5 ++++-
 configure.ac | 5 +++++
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index dcba62b..0a79319 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -7,6 +7,9 @@ pkgconfig_DATA = libdaq.pc
 if BUILD_MODULES
     MODULES_DIR = modules
 endif
-SUBDIRS = api $(MODULES_DIR) example test
+SUBDIRS = api $(MODULES_DIR) test
+if BUILD_EXAMPLE
+    SUBDIRS += example
+endif
 
 ACLOCAL_AMFLAGS = -I m4
diff --git a/configure.ac b/configure.ac
index 23b38a2..36a09ee 100644
--- a/configure.ac
+++ b/configure.ac
@@ -274,6 +274,11 @@ AM_CONDITIONAL([BUILD_MODULES], [test "$enable_afpacket_module" = yes -o \
 
 LIBS=${save_LIBS}
 
+AC_ARG_ENABLE(example,
+              AS_HELP_STRING([--disable-example],[do not build the example]),
+              [enable_example="$enableval"], [enable_example=yes])
+AM_CONDITIONAL([BUILD_EXAMPLE], [test "$enable_example" = yes])
+
 AC_CHECK_LIB([dl], [dlopen], [LIBDL="-ldl"])
 
 AM_CONDITIONAL([BUILD_SHARED_MODULES], [ test "$enable_shared" = yes ])
-- 
2.30.2

